src/Microsoft.Health.Fhir.CosmosDb.Core/Features/Storage/Versioning/CollectionUpgradeManager.cs

```diff
@@ -59,14 +67,8 @@ await using (var distributedLock = _lockFactory.Create(container, $"UpgradeLock:
                     _logger.LogDebug("Attempting to acquire upgrade lock");
 
                     await distributedLock.AcquireLock(cancellationTokenSource.Token);
-
-                    foreach (var updater in _collectionUpdater)
-                    {
-                        _logger.LogDebug("Running {CollectionUpdater} on {CollectionId}", updater.GetType().Name, _collectionConfiguration.CollectionId);
-
-                        await updater.ExecuteAsync(container, CancellationToken.None);
-                    }
-
+                    await _storedProcedureInstaller.ExecuteAsync(container, cancellationToken);
```

Comment:
I'm in doubt if we should continue calling storedProcedureInstaller here, or if we should move it to DataPlaneCollectionSetup.

Reply:
I was reviewing our plan for ArmCollectionSetup and I noticed that we'll create the stored procedures from CreateCollectionAsync. 
And we should probably be the same here. 
In that case, DataPlaneCollectionSetup should now have a dependency to IStoredProcedureInstaller. And we should remove IStoredProcedureInstaller from CollectionUpgradeManager.

Reply:
done